Development

noordean opened the pull request.

noordean committed:
Protect the exposed routes

Include groupID in api/group endpoint's response

Include groupID in api/group endpoint's response

Fix travis build error

Fix travis build issue

noordean committed:
Protect the exposed routes

noordean committed:
resolve merge conflict

noordean committed:
resolve merge conflict

noordean committed:
Add controller folder

noordean committed:
redesign the templates

noordean committed:
remove the dist contents

noordean committed:
refactor the API

- change the tests from jasmine to mocha
- write tests for model
- separate validations from routes

[Delivers]

noordean committed:
Add template and server to gitignore

noordean committed:
Add client and template to gitignore

noordean committed:
Add client folder to gitignore

noordean committed:
specify the gulp-mocha version in package.json

noordean committed:
fix axios request error

noordean committed:
fix API error

noordean committed:
fix API error of returning 404

noordean committed:
refactor users actions

noordean committed:
fix too many connections error

noordean committed:
put user's token in localstorage

noordean committed:
implement redirection after login

noordean committed:
Add travis database url

noordean committed:
Add travis database url

noordean committed:
fix test failing error

noordean committed:
add timeout to mocha test

noordean committed:
implement user logout

noordean committed:
setup a modal for creating group

noordean committed:
change mocha from gulp to npm

noordean committed:
include description to api/group endpoint

noordean committed:
validate api/create from the front-end

noordean committed:
connect group states to header componenet

noordean committed:
fix API connection error

noordean committed:
change the database_url

noordean committed:
Fix API response error

noordean committed:
add coverage command to travis

noordean committed:
fix logout error

noordean committed:
add api/users get request

noordean committed:
implement group creation from the modal

noordean committed:
add api/groups/:username to get user's groups

noordean committed:
reload page after group creation

noordean committed:
create a messageboard component

noordean committed:
fix sidenav issue

noordean committed:
add getMessages action

noordean committed:
resolve headers settings

noordean committed:
add header settings

noordean committed:
fix headers issue

noordean committed:
fix headers issue

noordean committed:
fix hidden sidenav

noordean committed:
add endpoints to get groups by id and add an array of users

noordean committed:
implement posting of message

noordean committed:
implement all message board functionalities

noordean committed:
extend the allowed number of characters in message field

noordean committed:
remove sync force:true

noordean committed:
convert the messages date to a readable format

noordean committed:
create pagination.js component

noordean committed:
add offset to api/groups/:username

noordean committed:
add loadGroupsHandler to pagination component

noordean committed:
add limit to groups query

noordean committed:
add limit to groupAction's endpoint

noordean committed:
make 'limit' optional in group query

noordean committed:
implement pagination on dashboard

noordean committed:
add more endpoints

noordean committed:
implement pagination for dashboard and messageboard

noordean committed:
refactor the actions and reducers

noordean committed:
test: write userReducers test

noordean committed:
resolve merge conflict

noordean committed:
create the missing index.js

noordean committed:
Convert the sidenav to fix

noordean committed:
fix

noordean committed:
Create models and association with migration

noordean committed:
create migration for UserGroup intermediate table

noordean committed:
Implement the needed routes

noordean committed:
Refactoring: Implement association and migration

- install sequelize cli
- create .sequelizerc file
- add new models, controllers and services

[Delivers]

noordean committed:
Refactor the server tests

noordean committed:
Merge pull request #29 from noordean/chore/refactor-migration-association

Chore/refactor migration association

noordean committed:
Merge branch 'server-side' of https://github.com/noordean/PostIt into chore/refactor-test
update the codebase

noordean committed:
Merge pull request #30 from noordean/chore/refactor-test

Chore/refactor test

noordean committed:
increase the test timeout

noordean committed:
Update travis.yml

noordean committed:
Update travis.yml

noordean committed:
install nyc to run mocha

noordean committed:
Fix the failing test

noordean committed:
Merge branch 'server-side' of https://github.com/noordean/PostIt into chore/refactor-test
fix the failing test
pick 1b0e9a9 style(formatting, coding style ...) Configure Eslint

noordean committed:
Merge pull request #31 from noordean/chore/refactor-test

Chore/refactor test

noordean committed:
add .nycrc file

noordean committed:
Merge branch 'server-side' of https://github.com/noordean/PostIt into server-side
Update server-side branch

noordean committed:
stabilize the client side

noordean committed:
fix logger

noordean committed:
chore(code-refactor): refactor the server side

- create helper function and middlewares to make controllers more readable
- add version v1 to the routes
- change limit and offset to query instead of params
- write more tests for more coverage

[Delivers]

noordean committed:
chore(code-refactor): refactor the client side

- fix page reloading
- combine reducer files together
- add prop-types to components
- add spread operator in the reducers
- separate pagination into its component
- change the stateless components to arrow functions

[Delivers]

noordean committed:
chore(code-refactor): update README file

- update readme file

[Delivers]

noordean committed:
chore(code-refactoring): refactor react client side

- set token in axios global headers
- re-write pagination into a different component
- combine all user, group and message reducers together

[Delivers]

noordean committed:
chore(code-refactoring): refactor react client side

    - set token in axios global headers
    - re-write pagination into a different component
    - combine all user, group and message reducers together

    [Delivers]

noordean committed:
chore(integration): integrate houndCI

- add .hound.yml file
- add .eslintignore

[Delivers #150354662]

noordean committed:
chore(integration): integrate houndCI

        - add .hound.yml file
        - add .eslintignore

        [Delivers #150354662]

noordean committed:
Merge pull request #32 from noordean/chore/#150354662/integrate-houndci

integrate houndci

noordean committed:
feat(implementation): implement forget password feature

    - create component, action and reducer called passwordReset
    - install nodemailer
    - create two controllers and routes to send mail and verify the sent url

    [Delivers]

noordean committed:
Merge pull request #33 from noordean/feature/#150570014/user-reset-password

user reset password

noordean committed:
feature(implementation): implement google signup

- add the GoogleLogin component
- create an app with google API
- create a route to register users through google

[Delivers #150353021]

noordean committed:
bug(bug-fix): fix travis

- install babel-plugin-transform-es2015-destructuring to fix nodemailer spread operator

[Delivers]

noordean committed:
Merge pull request #34 from noordean/feature/#150353021/google-signup

implement google signup

noordean committed:
feature(implementation): implement email notification for urgent and critical messages

- create a controller to send mail notification
- add an action, and a reducer for sending mail notification

[Delivers #150353653]

noordean committed:
Merge pull request #35 from noordean/feature/#150353653/email-notification

Implement Email Notification

hassanoloye commented on client/components/googleLogin.jsx:
```diff
@@ -0,0 +1,69 @@
+    return (
+      <GoogleLogin
+        className='google-login'
+        clientId='865334857353-2cgfcv29lanb6k7rsdo4qubn6fp1dvm6.apps.googleusercontent.com'
```
This should not be hardcoded. You could probably use environment variables to hold the values

hassanoloye commented on client/index.js:
```diff
@@ -1,25 +1,37 @@
+      <Route path="signup" component={Signup} />
+      <Route path="signin" component={Signin} />
+      <Route path="dashboard" component={Dashboard} />
+      <Route path="message-board/:groupID/:groupName" component={MessageBoard} />
```
Any reasons why you have more than one url params here?
I think we could have
`<Route path="message-board/:groupID/" component={MessageBoard} />`
or
`<Route path="message-board/:groupName" component={MessageBoard} />`

Alternatively, you can include a slug in your group model and use the slug in the url like this 
`<Route path="message-board/:groupSlug" component={MessageBoard} />`

hassanoloye commented on client/test/actions/user.spec.js:
```diff
@@ -0,0 +1,218 @@
+
+  describe('Signup Action', () => {
+    it('Should make a post request to sign in users', (done) => {
+      moxios.stubRequest('https://postit-api.herokuapp.com/api/user/signup', {
```
I think it will be ideal to have the api host in a variable. So, in case the url changes, the work to be done would be less as you would not have to replace in every places where you have the former url 
Maybe something like this.

```
describe('Signup Action', () => {
    const apiUrl = 'https://postit-api.herokuapp.com'
    it('Should make a post request to sign in users', (done) => {
      moxios.stubRequest(`${apiUrl}/api/user/signup`, {

     })
    })
})

hassanoloye commented on client/test/actions/user.spec.js:
```diff
@@ -0,0 +1,218 @@
+    });
+    it("Should dispatch appropraite action type if there's an unexpected error",
+      (done) => {
+        moxios.stubRequest('https://postit-api.herokuapp.com/api/user/signup', {
```
Same as above ☝️

hassanoloye commented on server/controllers/user.js:
```diff
@@ -0,0 +1,238 @@
+ * @return {Object} response containing all users
+ */
+  static getAllUsers(req, res) {
+    const userrs = req.headers.userrs;
```
Was there a mistake in this variable name. I would suggest changing this to something straightforward and explanatory

hassanoloye reviewed (commented):
Nice work so far. Try fix the suggested changes. 
In cases where the suggested changes occurred more than once but was only commented on once, try to make the changes in all occurrences.

And also try to add new lines at the end of all your files to remove the `No newline at end of file` github warnings.

noordean committed:
bug(bug-fixing): fix travis

            - change node version in travis.yml to 6

            [Delivers]

noordean committed:
Merge pull request #37 from noordean/bug/fix-travis-failing

 fix failing travis

noordean committed:
feature(implementation): implement sms notification feature

- install node-jusibe
- add sendSmsNotification controller

[Delivers #150579834]

noordean committed:
Merge pull request #38 from noordean/feature/#150579834/send-sms-notification

feature(implementation): implement sms notification feature

noordean committed:
chore(update): update README file

- document the API

[Delivers]

noordean committed:
Merge pull request #39 from noordean/chore/update-reamdme

Update README file

noordean committed:
update(update): update readme file

- update the readme file

[Delivers]

noordean committed:
Merge pull request #40 from noordean/chore/update-reamdme

Update reamdme file

noordean committed:
bug(bug-fix): remove logger

- remove redux console logger

[Deliver]

noordean committed:
Merge pull request #42 from noordean/bug/remove-loggers

Remove redux logger

noordean committed:
feature(implementaion): message archive feature

- add two routes to archive and get archived messages
- create a component to display archived messages

[Delivers #150353872]

noordean committed:
Merge pull request #43 from noordean/feature/#150353872/archive-message

Archive read messages

noordean committed:
feature(implementation): users should see those that have read a message

- create a route to get those that have read a message

[Delivers #150353772]

noordean committed:
Merge pull request #44 from noordean/feature/#150353772/see-those-that-have-a-message

Users should see those that have read a message

noordean committed:
feature(implementation): Send in-app notification to group members

    - create two routes that save and get notification from server
    - add materialize toast to display the notificatios

    [Delivers #150579898]

noordean committed:
Merge pull request #45 from noordean/feature/#150579898/implement-in-app-notification

Send in-app notification to group members
